dts: Add base binding for interrupt-controller, clock, gpio, gpio-nexus, and fixup pwm base

galak opened the pull request:
Add base bindings "classes" for interrupt-controller, clock, gpio, and gpio-nexus.  Update the pwm base to match style of the new base bindings and cleanup some additional properties in pwm.

galak committed:
dts: Add base interrupt-controller to dts bindings

Introduce a intc.yaml that interrupt controller bindings should inherit
from.  intc.yaml defines the properties "interrupt-controller" and
"#interrupt-cells" which all interrupt controllers should have.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
scripts/dts/gen_defines.py: Ignore 'interrupt-controller' prop

We don't want any defines generated for the boolean
'interrupt-controller'.  So skip it in write_props if we see it.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
dts: Add base gpio-controller to dts bindings

Introduce a gpio.yaml that GPIO controller bindings should inherit
from.  gpio.yaml defines the properties "gpio-controller" and
"#gpio-cells" which all gpio controllers should have.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
scripts/dts/gen_defines.py: Ignore 'gpio-controller' prop

We don't want any defines generated for the boolean
'gpio-controller'.  So skip it in write_props if we see it.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
dts/bindings: Add required #pwm-cells property to pwm base

All pwm controllers should have #pwm-cells so add it to the base pwm
binding.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

ioannisg commented on dts/bindings/interrupt-controller/intc.yaml:
```diff
@@ -0,0 +1,23 @@
+title: Interrupt Controller Base Structure
+
+description: >
+    This binding gives the base structures for all Interrupt Controllers devices
```
base structure
Controller devices

galak replied:
Fixed.

ioannisg reviewed (commented):
Looks ok to me

galak committed:
dts/bindings: Update pwm base binding

* Change pwm device bindings to include both base and pwm.yaml.  This
  allow for flexibility for any nodes that might not need/utilize the
  base binding.

* Added pwm.yaml to a few device bindings that were missing it.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
dts/bindings: Add base clock controller to dts bindings

Introduce a clock.yaml that clock controller bindings should inherit
from.  clock.yaml defines the properties "#clock-cells" which all
clock controllers should have.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
dts/bindings: Add gpio-nexus base binding

Add a base binding for GPIO nexus nodes.  This will get used by an GPIO
connectors.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

ulfalizer reviewed (approved):
Looks fine to me.

ulfalizer commented:
Maybe `gpio-controller.yaml` would be clearer than `gpio.yaml`, etc.

galak commented:
> Maybe `gpio-controller.yaml` would be clearer than `gpio.yaml`, etc.

Changed gpio and interrupt to be gpio-controller.yaml & interrupt-controller.yaml
